conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -225,6 +237,7 @@ private MethodSpec createFromObject(Collection<EnrichedField> enrichedFields) {
                 .addModifiers(Modifier.PUBLIC)
                 .returns(builderClass)
                 .addParameter(objectClass, "other")
+                .addCode(verifyNotBuilt())
```

Comment:
It's a bit concerning that this had to be added in so many places: granted there's probably not much change that's happening in this codepath, but maybe there's something we could refactor here?

I don't see any shared interfaces in javapoet, but maybe something like:

```
        return guardNotBuilt(
                        MethodSpec.methodBuilder("from")
                                .addModifiers(Modifier.PUBLIC)
                                .returns(builderClass)
                                .addParameter(objectClass, "other")::addCode,
                        assignmentBlock)
                .addStatement("return this")
                .build();
    }

    private <T> T guardNotBuilt(Function<CodeBlock, T> addCode, CodeBlock codeBlock) {
        addCode.apply(verifyNotBuilt());
        return addCode.apply(codeBlock);
    }
```

There's probably a nicer way to do this.

Reply:
How about this: I think we can use the fact that javapoet builders are not really staged builders:

```
  private MethodSpec createBuild(Collection<EnrichedField> enrichedFields, Collection<FieldSpec> fields) {
        MethodSpec.Builder method = newSetter("build")
                .addModifiers(Modifier.PUBLIC)
                .returns(objectClass)
                .addStatement("this.$N = true", BUILT_FIELD);

        if (enrichedFields.stream().filter(EnrichedField::isPrimitive).count() != 0) {
            method.addStatement("validatePrimitiveFieldsHaveBeenInitialized()");
        }

        return method.addStatement("return new $L", Expressions.constructorCall(objectClass, fields))
                .build();
    }

    private MethodSpec.Builder newSetter(String name) {
        return MethodSpec.methodBuilder(name).addCode(verifyNotBuilt());
    }
```

Reply:
The custom builder-factory method sounds nice -- I'd avoided doing anything special since the risk is it makes the code a little harder to follow, but becomes easier to miss validation in future changes. I think `newSetter` hits the sweet spot.

Reply:
This would be more involved than I thought because we generate the method signature for some methods using `BeanBuilderAuxiliarySettersUtils` functions, and those are reused between interfaces and the implementation. We can share it, but I'm not sure that the maintainability trade-offs would make sense in that case.